src/main/java/software/amazon/awssdk/crt/eventstream/ClientConnection.java

```diff
@@ -0,0 +1,197 @@
+package software.amazon.awssdk.crt.eventstream;
+
+import software.amazon.awssdk.crt.CRT;
+import software.amazon.awssdk.crt.CrtResource;
+import software.amazon.awssdk.crt.CrtRuntimeException;
+import software.amazon.awssdk.crt.io.ClientBootstrap;
+import software.amazon.awssdk.crt.io.ClientTlsContext;
+import software.amazon.awssdk.crt.io.SocketOptions;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Wrapper around an event stream rpc client initiated connection.
+ */
+public class ClientConnection extends CrtResource {
```

Comment:
Should we name this `EventStreamRpcClientConnection`?

in the `software.amazon.awssdk.crt.http` package, we still prefix the name of all classes like `HttpHeaders` and `HttpClientConnection`. This seemed like a good idea because there are many kinds of Headers and ClientConnections in this world. But `Http` is only 4 letters, and `EventStreamRpc` is 14

Reply:
Also, I don't see the "Rpc" being using anywhere in the java code, to distinguish it from pure "event-loop" protocol